compiler/luci/pass/include/luci/Pass/QuantizeWeightsWithGPTQPass.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __LUCI_QUANTIZE_WEIGHTS_WITH_GPTQ_PASS_H__
+#define __LUCI_QUANTIZE_WEIGHTS_WITH_GPTQ_PASS_H__
+
+#include <loco.h>
+
+#include <logo/Pass.h>
+
+#include <luci/Pass/QuantizationParameters.h>
+#include <luci/IR/CircleNode.h>
+#include <unordered_map>
+
+namespace luci
+{
+
+/**
+ * @brief Pass to quantize weights
+ */
+class QuantizeWeightsWithGPTQPass : public logo::Pass
+{
+public:
+  struct Context
+  {
+    loco::DataType input_model_dtype = loco::DataType::Unknown;
+    loco::DataType output_model_dtype = loco::DataType::Unknown;
+    QuantizationGranularity granularity = QuantizationGranularity::ChannelWise;
+    std::vector<LayerInfo> layers_info;
+  };
+
+public:
+  QuantizeWeightsWithGPTQPass(std::unique_ptr<Context> &&ctx) : _ctx{std::move(ctx)}
+  {
+    // DO NOTHING
+  }
+
+  QuantizeWeightsWithGPTQPass(
+    std::unique_ptr<Context> &&ctx,
+    std::unordered_map<const luci::CircleNode *, std::vector<float>> *hessian_map)
```

Comment:
maybe use alias like in https://github.com/Samsung/ONE/pull/14265/files#diff-ad46570fba7cec76f781b0f9e6a570b9ddbfd1151a7747f0c40e094f81e7a10bR28 ?